stan/math/prim/mat/prob/ordered_logistic_lpmf.hpp

```diff
@@ -67,31 +75,180 @@ namespace stan {
       check_bounded(function, "Random variable", y, 1, K);
       check_finite(function, "Location parameter", lambda);
       check_greater(function, "Size of cut points parameter", c.size(), 0);
-      for (int i = 1; i < c.size(); ++i)
-        check_greater(function, "Cut points parameter", c(i), c(i - 1));
+      check_ordered(function, "Cut-points", c);
+      check_finite(function, "Final cut-point", c(c.size()-1));
+      check_finite(function, "First cut-point", c(0));
+
+      if (y == 1) {
+        logp_n += -log1p_exp(lambda - c[0]);
```

Comment:
You can just use `-=` here and in other cases--- I didn't catch this the first time.  No big deal if you don't want to change it.